metals/src/main/scala/scala/meta/internal/metals/WorkspaceSymbolProvider.scala

```diff
@@ -203,12 +205,16 @@ final class WorkspaceSymbolProvider(
       symbol <- index.symbols
       if query.matches(symbol.symbol)
     } {
-      visitor.visitWorkspaceSymbol(
-        path,
-        symbol.symbol,
-        symbol.kind,
-        symbol.range,
-      )
+      count += 1
+      if (query.isExact && count >= MaxWorkspaceMatchesForShortQuery) return
```

Comment:
Not super fond of the return. The alternative is to use a recursion after the for comprehension that would return an iterator.

Something like:
```scala
val symbols: Iterator[Symbol] = for {...}

@tailrec
def loop(currentNum: Int): Unit = {
  if ( ! (query.isExact ||  count >= MaxWorkspaceMatchesForShortQuery) && symbols.hasNext) 
    val sym = symbols.next()
    visitor.visitWorkspaceSymbol(
          path,
          symbol.symbol,
          symbol.kind,
          symbol.range,
        )
     loop(currentNum + 1)
   }
}
```
might not be exactly right.